main.go

```diff
@@ -493,9 +494,69 @@ func clean(state *state.HelmState, errs []error) error {
 
 func getArgs(c *cli.Context, state *state.HelmState) []string {
 	args := c.String("args")
+	argsMap := map[string]string{}
+
 	if len(args) > 0 {
-		state.HelmDefaults.Args = strings.Split(args, " ")
+		argsVals := strings.Split(args, " ")
+		for _, arg := range argsVals {
+			argVal := strings.SplitN(arg, "=", 2)
+			if len(argVal) > 1 {
+				arg := argVal[0]
+				value := argVal[1]
+				argsMap[arg] = value
+			} else {
+				arg := argVal[0]
+				argsMap[arg] = ""
+			}
+		}
 	}
+	if len(state.HelmDefaults.Args) > 0 {
+		for _, arg := range state.HelmDefaults.Args {
+			argVal := strings.SplitN(arg, "=", 2)
+			arg := argVal[0]
+			if _, exists := argsMap[arg]; !exists {
+				if len(argVal) > 1 {
+					argsMap[arg] = argVal[1]
+				} else {
+					argsMap[arg] = ""
+				}
+			}
+		}
+	}
+
+	setDedicatedKeys(state, argsMap)
```

Comment:
Shouldn't we move this to line 498? Otherwise the `--args` can be overwrote by `helmDefaults`, which seems to be the opposite of what I expect? 🤔

Reply:
Never mind. You do guard against it at https://github.com/roboll/helmfile/pull/186/files#diff-7ddfb3e035b42cd70649cc33393fe32cR557!